tests/Tests.elm

```diff
@@ -283,6 +283,13 @@ all =
                     \() ->
                         Expect.equal (Just Medium) <|
                             getDifficulty (List.repeat 2 3) (List.repeat 2 100)
+
+                , test "15 rugs of smothering are Deadly for 2 level 20s" <|
+                    -- 5x multiplier: 5(450xp x 15) >= 25400 Deadly
+                    \() ->
+                        Expect.equal (Just Deadly) <|
+                            getDifficulty (List.repeat 2 20) (List.repeat 15 450)
```

Comment:
This block of code is an [elm-test] test case. The initial comma indicates a new list item will follow, as the tests are grouped logically in nested lists (you can see that in the full file).

E.g.
```elm
[ test "first test" ...
, test "second test" ...
, ...
] -- this square bracket indicates the end of the list. Note no comma should immediately precede it.
```

The `test` function takes an initial argument of a quoted string describing the test case. The chevron and pipe symbols at the end of the line indicate that the (uncommented) line is the second argument for the `test` function, the actual test code itself. The following example lays that out, including the comment syntax at the end of the line (`--`).
```elm
test "this test does something" <| -- test code comes after that symbol
```
In the commit, my comment on the next line is just a shorthand description of the situation being tested. It's not required, I'm just putting that in to keep track of the encounter details.

The next three lines are the syntax for describing code that will actually run the test. You should just copy this and only change three parts: `Just Deadly` on line 290, `List.repeat 2 20` and `List.repeat 15 450` on 291. **Note:** Elm requires precise indentation to determine the context of code. If you copy the example, it should be fine (4 spaces per level of indentation, **no tab characters**).

- `Just Deadly` indicates the expected results of the test, which is the Deadly difficulty type. There are four options, corresponding to what's in the DMG: `Deadly`, `Hard`, `Medium`, and `Easy`. Any of those values is valid and accepted.
- The first `List.repeat` is a function that generates a list of *party character levels*. The two numbers that follow the function name are the number of list items to generate and the value to be repeated in the list. Our example of `List.repeat 2 20` generates `[ 20, 20 ]` corresponding to two 20th level party members.
- The second `List.repeat` call generates a list of *monster XP values*. The syntax is the same, thus in the example `List.repeat 15 450` we created a list of 15 instances of 450 (much too long to type out in full list literal format), which represents an encounter of 15 monsters, all worth 450xp.

If anyone gets this far and has the ambition to learn to make asymmetrical encounters, I can give some `List` function examples for cases where literal lists would be too long.

[elm-test]: http://package.elm-lang.org/packages/elm-community/elm-test/latest